src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -103,3 +109,29 @@ object GenomeReference {
   } yield GenomeReference(name, contigs, Set(xContig), Set(yContig), Set(mtContig), parX ++ parY)
 }
 
+case class GRVariable(gr: GenomeReference,  var t: Type = null) extends Serializable {
```

Comment:
It should extend `GenomeReference`.  Deciding the right way to do this might be a bit tricky.  Then TVariant.unify, etc., should unify the genome reference variable.  Let me know if this doesn't make sense.